internal/controllers/rpc.go

```diff
@@ -55,11 +55,31 @@ func (c ApiController) RPCHandler(w http.ResponseWriter, r *http.Request) {
 		)
 		if err != nil {
 			log.Errorf("Request failed to node %s because of: %v", node.ID, err)
-			go record.FailedRequest(node, c.nodeRepo, c.recordRepo)
+			// start penalize node action
+			go c.actions.PenalizeNode(node, c.repositories)
```

Comment:
This still made more sense to me if it was wrapped in some function like before like FailedRequest (PenalizeNode, whatever...).
Right now the code in controller is larger and also we wait for saving in database of record repo which I think should not be in same goroutine.